pkg/operator/ceph/disruption/machinelabel/reconcile.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package machinelabel
+
+import (
+	"context"
+
+	"github.com/coreos/pkg/capnslog"
+	mapiv1 "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/operator/ceph/disruption/controllerconfig"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	controllerName                  = "machinelabel-controller"
+	MachineFencingLabelKey          = "fencegroup.rook.io/cluster"
```

Comment:
If we are removing usage of fencing, here might be another place to change. `disruption-budget.rook.io/cluster` or `machine-disruption.rook.io/cluster`.

It's also probably a good idea to follow the pattern of separating words by a hyphen as has been done with the failure domain labels: https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/

Reply:
- Fencing is still a valid term for the overall feature. It's described as such in the [design doc](https://github.com/rook/rook/blob/master/design/ceph-openshift-fencing-mitigation.md). I think the clarification is useful where we really were disabling the controller for MDBs. The MDBs are essentially the resource type that enables the fencing feature. So keeping fencing here makes sense when we're not referring specifically to the MDB resource or the controller.
- For the second point, I'd rather not rename the label at this point and reset testing.